src/draw.js

```diff
@@ -91,47 +108,136 @@ Draw.prototype = extend(Control, {
         if (!this.squareCtrl.classList.contains('active')) {
           this.squareCtrl.dispatchEvent(event);
         }
-      break;
+        break;
+      case 27: // (escape) exit edit mode
+        if (this.editId) {
+          this._exitEdit();
+        }
+        break;
     }
   },
 
-  _onMouseDown(e) {
-    if (e.altKey) {
-      e.stopPropagation();
-      // TODO https://github.com/mapbox/mapbox-gl-js/issues/1264
-      // this._captureFeatures();
+  _onMouseDown(/*e*/) {
+    //if (e.altKey) {
+    //  e.stopPropagation();
+    //    TODO https://github.com/mapbox/mapbox-gl-js/issues/1264
+    //    this._captureFeatures();
+    //}
+  },
+
+  _onClick(e) {
+    var coords = DOM.mousePos(e, this._map._container);
+    this._map.featuresAt([coords.x, coords.y], { radius: 10 }, (err, features) => {
+      if (err) throw err;
+      else if (!features.length && this.editId) return this._exitEdit();
+      else if (!features.length) return;
+
+      var feature = features[0];
+      if (this.editId && this.editId !== feature.properties._drawid) this._exitEdit();
+      else if (this.editId === feature.properties._drawid) return;
+
+      this.editId = feature.properties._drawid;
+      coords = feature.geometry.coordinates;
+      var type = feature.geometry.type;
+
+      if (type === 'Point')
+        this.featureType = 'point';
+      else if (type === 'LineString' || type === 'MultiLineString')
+        this.featureType = 'line';
+      else if (coords[0][0][0] === coords[0][1][0])
+        this.featureType = 'square';
+      else
+        this.featureType = 'polygon';
+
+      this._edit();
+    });
+  },
+
+  _edit() {
+    // move the feature clicked on out of the draw store into an edit store
+    var activeFeature = this.options.geoJSON.edit(this.editId);
+    this.editStore = new EditStore([activeFeature]);
+    this._map.fire('draw.feature.update', { geojson: this.options.geoJSON.getAll() });
+    this._map.fire('edit.feature.update', { geojson: this.editStore.getAll() });
+    this._map.getContainer().addEventListener('mousedown', this.initiateDrag, true);
+  },
+
+  _exitEdit() {
+    // save the changes into the draw store
+    this.options.geoJSON.save(this.editStore.getAll());
+    this._map.fire('draw.feature.update', { geojson: this.options.geoJSON.getAll() });
+    this.editStore.clear();
+    this._map.fire('edit.feature.update', { geojson: this.editStore.getAll() });
+    this._map.getContainer().removeEventListener('mousedown', this.initiateDrag, true);
+    this.editId = false;
+  },
+
+  _initiateDrag() {
+    this._map.getContainer().addEventListener('mousemove', this.drag, true);
+    this._map.getContainer().addEventListener('mouseup', this.endDrag, true);
+  },
+
+  _drag(e) {
+    e.stopPropagation();
+    if (!this.dragging) {
+      this.dragging = true;
+      this.init = DOM.mousePos(e, this._map.getContainer());
+      var options = R.merge(this.options, { geoJSON: this.editStore });
```

Comment:
oops, maybe, idk. I used `R.merge` here to replace that key in the object with this new value.